### ISSUE TITLE
`@stylistic/string-quotes` drops a comment standing in the selector when it fixes an attribute value

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#178]
Found while fixing [#170](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/170), which teaches the plugin to read a selector whose quoted attribute value holds an interpolation. The loss below has nothing to do with interpolation and is reproducible on `main`.

### What minimal example or steps are needed to reproduce the bug?

```css
.foo /* x */ [title="y"] {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/string-quotes": "single"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

```css
.foo /* x */ [title='y'] {}
```

### What actually happened?

The comment is gone, and the whitespace it stood in is left behind:

```css
.foo  [title='y'] {}
```

The warning that is fixed points at the wrong character too. Without `--fix` the same selector is reported at `1:14`, where the file has the `*` of the comment's closing delimiter; the quotation mark it is about stands at `1:21`.

### Do you have a proposal to fix the bug?

PostCSS keeps two spellings of a selector holding a comment: the text the file has, in `rule.raws.selector.raw`, and a copy with every comment taken out, in `rule.raws.selector.value` — and `rule.selector` gives back the second one.

```js
postcss.parse(`.foo /* x */ [title="y"] {}`).first.raws.selector
// { raw: '.foo /* x */ [title="y"]', value: '.foo  [title="y"]' }
```

`string-quotes` reads `ruleNode.selector`, so the selector it parses never held the comment — hence the position off its mark — and it writes the fix back with `ruleNode.selector = selectorTree.toString()`, which discards the raw along with the comment.

The shape this needs is the one `selector-descendant-combinator-no-non-space` and `selector-pseudo-class-parentheses-space-inside` already have after [#124](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/124) and [#163](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/163): read `ruleNode.raws.selector.raw` where there is one, count every position in it, and write the fix to the raw rather than to `ruleNode.selector`.

Of the rules that write a selector, this one is alone in losing the comment: `selector-attribute-brackets-space-inside`, `selector-pseudo-class-case`, `selector-list-comma-space-after`, `selector-max-empty-lines` and `selector-descendant-combinator-no-non-space` all keep it, each on the same `.foo /* x */ …` shape.
